crates/release_plz_core/src/next_ver.rs

```diff
@@ -372,6 +385,25 @@ pub fn next_versions(input: &UpdateRequest) -> anyhow::Result<(PackagesUpdate, T
         project: &local_project,
         req: input,
     };
+
+    let repository = local_project
+        .get_repo()
+        .context("failed to determine local project repository")?;
+    if !input.allow_dirty {
+        repository.repo.is_clean()?;
+    }
+
+    // modify the input for `git-only` case by creating a new copy of it
+    let input = if input.packages_config.default.git_only && input.registry_manifest.is_none() {
+        let registry_manifest_path = fetch_registry_manifest_from_git(&repository, &input)
+            .context("failed to fetch registry manifest from git")?;
+        let mut new_input = input.clone();
+        new_input.registry_manifest = Some(registry_manifest_path);
+        new_input
+    } else {
+        input.clone()
+    };
```

Comment:
I'm not a fan of this approach (editing the input).
Is there another approach? 🤔 
Also, what if users want to use both git_only and registry_manifest

Reply:
> Is there another approach? 🤔

Sure, no worries!

> Maybe we should document this or add a check when we read the configuration. I.e. if git_only is true and registry_manifest is specified, throw an error.

Exactly! I was saving this discussion for later to not bloat the current task with details, glad we are already on the same page about that one 👍 

Will write the changes soon.

Reply:
Addressed in [98be190](https://github.com/MarcoIeni/release-plz/pull/1497/commits/98be19003268f9475a29c844c7d7ff33569fb082)